kythe/java/com/google/devtools/kythe/platform/java/JavacOptionsUtils.java

```diff
@@ -52,18 +60,64 @@
  */
 public class JavacOptionsUtils {
 
+  private static final Path JAVA_HOME = Paths.get(StandardSystemProperty.JAVA_HOME.value());
   private static final ImmutableList<String> JRE_JARS =
-      ImmutableList.of(
-          "lib/rt.jar", "lib/resources.jar", "lib/jsse.jar", "lib/jce.jar", "lib/charsets.jar");
+      ImmutableList.of("rt.jar", "resources.jar", "jsse.jar", "jce.jar", "charsets.jar");
   private static final ImmutableList<String> JRE_PATHS;
 
   static {
+    try {
+      ImmutableList<String> paths = getBootClassPath8();
+      if (paths.isEmpty()) {
+        paths = getBootclassPath();
+      }
+      JRE_PATHS = paths;
+    } catch (IOException ioe) {
+      throw new IllegalStateException(ioe);
+    }
+  }
+
+  /** Returns the JRE 8-style bootclasspath. */
+  private static ImmutableList<String> getBootClassPath8() throws IOException {
+    Path jreLib = JAVA_HOME.resolve("lib");
+    ImmutableList.Builder<String> jars = ImmutableList.builder();
+    Path extDir = jreLib.resolve("ext");
+    if (Files.exists(extDir)) {
+      try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(extDir, "*.jar")) {
+        for (Path extJar : dirStream) {
+          jars.add(extJar.toString());
+        }
+      }
+    }
+    for (String jar : JRE_JARS) {
+      Path path = jreLib.resolve(jar);
+      if (Files.exists(path)) {
+        jars.add(path.toString());
+      }
+    }
+    return jars.build();
+  }
+
+  /** Returns the JRE 9-style bootclasspath. */
+  private static ImmutableList<String> getBootclassPath() throws IOException {
+    Context context = new Context();
+    JavacTool.create()
+        .getTask(
+            /* out = */ null,
+            /* fileManager = */ null,
+            /* diagnosticListener = */ null,
+            /* options = */ Arrays.asList("--system", JAVA_HOME.toString()),
+            /* classes = */ null,
+            /* compilationUnits = */ null,
+            context);
+    StandardJavaFileManager fileManager =
+        (StandardJavaFileManager) context.get(JavaFileManager.class);
```

Comment:
Just: `javax.tools.JavaCompiler.getStandardFilemanager(null, null, null)`?

Reply:
Aside from that, it's odd that we're constructing a filemanager to ask it for the platform_class_path which we're then using to construct a string argument list used to configure the filemanager with the platform_class_path.

Reply:
Yeah, that works.